### PR TITLE
Add tests for `nn.functional.margin_ranking_loss` (dup)

### PR DESCRIPTION
**Pull request by nkaretnikov**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #72327
* #72326
* #72325
* #72324
* #72323
* #72322
* #72321
* __->__ #72320
* #72319
* #72318

`out` is not tested because other loss functions don't use it either.

Also see PR #67444.

### CODE REVIEW

**Comment by nkaretnikov** [comment]
going to merge this with other related changes

**Comment by nkaretnikov** [comment]
squashed to https://github.com/pytorch/pytorch/pull/72321
